### DISCUSSION TITLE
Create and add if does not exist

### DISCUSSION THREAD

**Discussion by electriquo** [TomWright/dasel#338] [Q&A]
how can i make dasel to create an array and add an object to it if the object is does not exist.
for instance, giving an empty json/yaml, how to make dasel create a key named `names` that is an array with the object `{"name": "Tom", "level": "Master"}` to achieve

```yaml
names:
  - name: Tom
    level: Master
```

yet, if the `{"name": "Tom", "level": "Master"}` object already exist, then do nothing.

**Comment by TomWright** [comment]
This kind of optional update doesn't really exist yet, but you should be able to combine 2 dasel commands together to get the expected outcome.

To check the presence of some item you could use this.
```
$ echo 'names:
  - name: Tom
    level: Master' | dasel -r yaml 'moreThan(names.all().filter(and(equal(name,Tom),equal(level,Master))).merge().len(),0)'
true
$ echo 'names:
  - name: Tom
    level: Master' | dasel -r yaml 'moreThan(names.all().filter(and(equal(name,Something),equal(level,Else))).merge().len(),0)'
false
```

You can then wrap this with a simple check and perform some action:
```
if dasel -f file.yaml 'moreThan(names.all().filter(and(equal(name,Tom),equal(level,Master))).merge().len(),0)' | grep -q 'true'
then
  echo "already exists"
else
  echo "adding"
  dasel put -f file.yaml -t json -v '{"name":"Tom","level":"Master"}' 'names.[]'
fi
```

**Reply by electriquo** [reply]
Any reference/guidance/example would be appreciated

**Reply by TomWright** [reply]
Hi @foolioo I've added a rough example. Note that I haven't tested this and it serves as my thought process, not necessarily a complete and working example.
